sw/device/silicon_creator/lib/drivers/otp_unittest.cc

```diff
@@ -0,0 +1,99 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/drivers/otp.h"
+
+#include "gtest/gtest.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/base/testing/mock_mmio.h"
+#include "sw/device/silicon_creator/lib/error.h"
+
+#include "otp_ctrl_regs.h"  // Generated.
+
+namespace otp_unittest {
+namespace {
+using ::mock_mmio::MmioTest;
+using ::testing::Test;
+
+class OtpTest : public Test, public MmioTest {
+ protected:
+  otp_t otp_ = {.base_addr = dev().region()};
+};
+
+class OtpReadTest : public OtpTest {
+ protected:
+  const std::vector<uint8_t> data_ = {
+    0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07, 0x08, 0x09, 0x0A, 0x0B,
+    0x0C, 0x0D, 0x0E, 0x0F,
+  };
+  const ptrdiff_t offset_ = OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET;
+
+  uint32_t get_word(const std::vector<uint8_t> &vec, ptrdiff_t i) {
+    auto v = vec.begin() + i;
+    return (v[0] << 24) | (v[1] << 16) | (v[2] << 8) | v[3];
```

Comment:
Out of curiosity, why bother w the vector when you can just synthetically generate this word?

E.g.

```cpp
uint32_t val = 0;
for (int j = 0; j < 4; ++j) {
  val <<= 8;
  val |= i + j;
}
return val;
```